docs/design/architecture.md

```diff
@@ -2,128 +2,353 @@
+### Workload root filesystem
 
-The `kata-agent` execution unit is the sandbox. A `kata-agent` sandbox is a container sandbox defined by a set of namespaces (NS, UTS, IPC and PID). `shimv2` can
-run several containers per VM to support container engines that require multiple
-containers running inside a pod.
+For details of how the [runtime](#runtime) make the workload image
+chosen by the user (such as `busybox:latest`) available to the
+workload process, see the [storage](#storage) section.
 
-`kata-agent` communicates with the other Kata components over `ttRPC`.
+The container workload process defined by the OCI bundle's
+configuration file (`config.json`) is created using the Kata
+Containers [agent](#agent). The agent creates a container environment
+to run the workload in, in a similar manner to that done by the
+[`runc`](https://github.com/opencontainers/runc) OCI runtime, using
+Linux cgroups and namespaces.
+
+Note that the workload is isolated from the VM root environment by its
+surrounding container environment. The VM environment the workload
+container runs in is also isolated from the "outer" host environment
+where the container manager runs.
 
 ## Runtime
 
-`containerd-shim-kata-v2` is a [containerd runtime shimv2](https://github.com/containerd/containerd/blob/v1.4.1/runtime/v2/README.md) implementation and is responsible for handling the `runtime v2 shim APIs`, which is similar to [the OCI runtime specification](https://github.com/opencontainers/runtime-spec) but simplifies the architecture by loading the runtime once and making RPC calls to handle the various container lifecycle commands. This refinement is an improvement on the OCI specification which requires the container manager call the runtime binary multiple times, at least once for each lifecycle command.
+The Kata Containers runtime (or shim), is a [shimv2](#shim-v2-architecture) compatible runtime.
 
-`containerd-shim-kata-v2` heavily utilizes the
-[virtcontainers package](../../src/runtime/virtcontainers/), which provides a generic, runtime-specification agnostic, hardware-virtualized containers library.
+The name of the Kata Containers shim binary is
+[`containerd-shim-kata-v2`](../../src/runtime/cmd/containerd-shim-kata-v2).
 
-### Configuration
+The runtime makes heavy use of the [`virtcontainers`
+package](../../src/runtime/virtcontainers), which provides a generic,
+runtime-specification agnostic, hardware-virtualized containers
+library.
+
+The runtime is responsible for starting the [hypervisor](#hypervisor)
+and it's VM, and communicating with the [agent](#agent) using a
+[ttRPC based protocol](#agent-communications-protocol) over a VSOCK
+socket that provides a communications link between the VM and the
+host. 
+
+This protocol allows the runtime to send container management commands
```

Comment:
I think another sentence explaining how this communication into the VM is secure would help reinforce Kata as a security isolation product.

Reply:
sounds good to me.